vms/rpcchainvm/ghttp/grequest/request_client.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package grequest
+
+import (
+	"context"
+	"io"
+
+	"github.com/ava-labs/avalanchego/proto/pb/http/request"
+)
+
+var _ io.Reader = (*Client)(nil)
+
+func NewClient(ctx context.Context, client request.RequestClient) *Client {
+	return &Client{
+		ctx:    ctx,
+		client: client,
+	}
+}
+
+type Client struct {
+	ctx    context.Context
+	client request.RequestClient
+}
+
+func (c *Client) Read(p []byte) (n int, err error) {
+	reply, err := c.client.Body(c.ctx, &request.BodyRequest{N: uint32(len(p))})
+	if err != nil {
+		return 0, err
+	}
```

Comment:
I think it's fine for us to just return the error here. `io.EOF` is a special error for both grpc and `io.Reader` ([ref](https://pkg.go.dev/io?utm_source=godoc#Reader)). `io.EOF`'s invariants on `io.Reader` are particularly confusing though.